core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -659,6 +677,35 @@ private[spark] class ExecutorAllocationManager(
     // place the executors.
     private val stageIdToExecutorPlacementHints = new mutable.HashMap[Int, (Int, Map[String, Int])]
 
+    override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+      var jobGroupId = if (jobStart.properties != null) {
+        jobStart.properties.getProperty(SparkContext.SPARK_JOB_GROUP_ID)
+      } else {
+        null
+      }
+
+      val maxConTasks = if (jobGroupId != null &&
+        conf.contains(s"spark.job.$jobGroupId.maxConcurrentTasks")) {
+        conf.get(s"spark.job.$jobGroupId.maxConcurrentTasks").toInt
+      } else {
+        Int.MaxValue
+      }
+
+      if (maxConTasks <= 0) {
+        throw new IllegalArgumentException(
+          "Maximum Concurrent Tasks should be set greater than 0 for the job to progress.")
+      }
+
+      if (jobGroupId == null || !conf.contains(s"spark.job.$jobGroupId.maxConcurrentTasks")) {
+        jobGroupId = DEFAULT_JOB_GROUP
+      }
```

Comment:
this can be simplified to

```scala
val jobGroupId = if (jobStart.properties != null) {
        Option(jobStart.properties.getProperty(SparkContext.SPARK_JOB_GROUP_ID)).getOrElse(DEFAULT_JOB_GROUP)
      } else {
        DEFAULT_JOB_GROUP
      }
  val maxConTasks = conf.getOption(s"spark.job.$jobGroupId.maxConcurrentTasks").map(_.toInt).getOrElse(Int.MaxValue)
```

Reply:
actually though I realize reading the rest of the comments, to make sure you get the properties from the job that actually ends up starting the stage, you shouldn't grab the properties here at all.  You should grab them from `stageSubmitted.properties`